pty/ptytest/ptytest.go

```diff
@@ -57,76 +75,68 @@ func create(t *testing.T, ptty pty.PTY, name string) *PTY {
 	out := newStdbuf()
 	w := io.MultiWriter(logw, out)
 
-	tpty := &PTY{
+	ex := &outExpecter{
 		t:    t,
-		PTY:  ptty,
 		out:  out,
 		name: name,
 
 		runeReader: bufio.NewReaderSize(out, utf8.UTFMax),
 	}
-	// Ensure pty is cleaned up at the end of test.
-	t.Cleanup(func() {
-		_ = tpty.Close()
-	})
 
 	logClose := func(name string, c io.Closer) {
-		tpty.logf("closing %s", name)
+		ex.logf("closing %s", name)
 		err := c.Close()
-		tpty.logf("closed %s: %v", name, err)
+		ex.logf("closed %s: %v", name, err)
 	}
 	// Set the actual close function for the tpty.
-	tpty.close = func(reason string) error {
+	ex.close = func(reason string) error {
 		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
 		defer cancel()
 
-		tpty.logf("closing tpty: %s", reason)
+		ex.logf("closing expecter: %s", reason)
 
-		// Close pty only so that the copy goroutine can consume the
-		// remainder of it's buffer and then exit.
-		logClose("pty", ptty)
+		// Caller needs to have closed the PTY so that copying can complete
 		select {
 		case <-ctx.Done():
-			tpty.fatalf("close", "copy did not close in time")
+			ex.fatalf("close", "copy did not close in time")
 		case <-copyDone:
 		}
 
 		logClose("logw", logw)
 		logClose("logr", logr)
 		select {
 		case <-ctx.Done():
-			tpty.fatalf("close", "log pipe did not close in time")
+			ex.fatalf("close", "log pipe did not close in time")
 		case <-logDone:
 		}
 
-		tpty.logf("closed tpty")
+		ex.logf("closed expecter")
 
 		return nil
 	}
 
 	go func() {
 		defer close(copyDone)
-		_, err := io.Copy(w, ptty.Output())
-		tpty.logf("copy done: %v", err)
-		tpty.logf("closing out")
+		_, err := io.Copy(w, r)
+		ex.logf("copy done: %v", err)
+		ex.logf("closing out")
 		err = out.closeErr(err)
-		tpty.logf("closed out: %v", err)
+		ex.logf("closed out: %v", err)
 	}()
 
 	// Log all output as part of test for easier debugging on errors.
 	go func() {
 		defer close(logDone)
 		s := bufio.NewScanner(logr)
 		for s.Scan() {
-			tpty.logf("%q", stripansi.Strip(s.Text()))
+			ex.logf("%q", stripansi.Strip(s.Text()))
 		}
 	}()
 
-	return tpty
+	return ex
 }
 
-type PTY struct {
-	pty.PTY
+type outExpecter struct {
```

Comment:
It feels like the convention in go is to call things by their methods, rather than the actions they take, e.g. `io.Reader` is something which has a "Read" method.  In this case, the struct exposes "expect" methods, so I called it "Expecter"

Reply:
By that reasoning, Read wins since there's only "one" expect method (the non-context one should go away) 😄.

<img width="161" alt="image" src="https://user-images.githubusercontent.com/147409/233336138-2ca8fd6c-5c39-4e84-a6b3-1e9e37cf7ef7.png">